### DISCUSSION TITLE
If I do not pass a variable into query key, and what will happen?

### DISCUSSION THREAD

**Discussion by AaronConlon** [TanStack/query#4963] [Q&A] [answered]
Hi there.I see this description in doc:
> [If your query function depends on a variable, include it in your query key](https://tanstack.com/query/v4/docs/vue/guides/query-keys#if-your-query-function-depends-on-a-variable-include-it-in-your-query-key)

So, I want to know.If I forget to pass a variable include into my query key and what will happen.Thanks.

**Comment by DamianOsipiuk** [comment, marked as answer]
In this case whenever this variable changes, your query would not update.
You could treat it as a dependency array like in other react hooks.
Additionally based on `queryKey`, cache entry is created. So if you do not include changing variable in `queryKey`, different queries might want to store data in the same place.
